cli/command/command.go

```diff
@@ -18,11 +18,72 @@ package command
 
 import (
 	"fmt"
+	"strings"
 
+	"github.com/Sirupsen/logrus"
 	"github.com/skippbox/kompose/cli/app"
 	"github.com/urfave/cli"
 )
 
+// Hook for erroring and exit out on warning
+type errorOnWarningHook struct{}
+
+func (errorOnWarningHook) Levels() []logrus.Level {
+	return []logrus.Level{logrus.WarnLevel}
+}
+
+func (errorOnWarningHook) Fire(entry *logrus.Entry) error {
+	logrus.Fatalln(entry.Message)
+	return nil
+}
+
+// BeforeApp is an action that is executed before any cli command.
+func BeforeApp(c *cli.Context) error {
+
+	if c.GlobalBool("verbose") {
+		logrus.SetLevel(logrus.DebugLevel)
+	} else if c.GlobalBool("suppress-warnings") {
+		logrus.SetLevel(logrus.ErrorLevel)
+	} else if c.GlobalBool("error-on-warning") {
+		hook := errorOnWarningHook{}
+		logrus.AddHook(hook)
+	}
+
+	provider := strings.ToLower(c.GlobalString("provider"))
+	switch provider {
+	case "kubernetes":
+		k8sControllers := []cli.Flag{
```

Comment:
Yes sure we can create new `cli.Command`, and nice catch of that usage string BTW!